services/s3/src/it/java/software/amazon/awssdk/services/SelectObjectContentIntegrationTest.java

```diff
@@ -0,0 +1,143 @@
+import software.amazon.awssdk.services.s3.model.CSVInput;
+import software.amazon.awssdk.services.s3.model.CSVOutput;
+import software.amazon.awssdk.services.s3.model.CompressionType;
+import software.amazon.awssdk.services.s3.model.ExpressionType;
+import software.amazon.awssdk.services.s3.model.InputSerialization;
+import software.amazon.awssdk.services.s3.model.OutputSerialization;
+import software.amazon.awssdk.services.s3.model.RecordsEvent;
+import software.amazon.awssdk.services.s3.model.SelectObjectContentEventStream;
+import software.amazon.awssdk.services.s3.model.SelectObjectContentEventStream.EventType;
+import software.amazon.awssdk.services.s3.model.SelectObjectContentRequest;
+import software.amazon.awssdk.services.s3.model.SelectObjectContentResponse;
+import software.amazon.awssdk.services.s3.model.SelectObjectContentResponseHandler;
+
+public class SelectObjectContentIntegrationTest extends S3IntegrationTestBase {
+    private static final String BUCKET_NAME = temporaryBucketName(SelectObjectContentIntegrationTest.class);
+    private static final String KEY = "test_object.csv";
+    private static final String CSV_CONTENTS = "A,B\n"
+                                               + "C,D";
+    private static final String QUERY = "select s._1 from S3Object s";
+
+
+    private static S3AsyncClient s3Async;
+
+    @BeforeAll
+    public static void setup() throws Exception {
+        S3IntegrationTestBase.setUp();
+        s3.createBucket(r -> r.bucket(BUCKET_NAME));
+        s3.waiter().waitUntilBucketExists(r -> r.bucket(BUCKET_NAME));
+        s3.putObject(r -> r.bucket(BUCKET_NAME).key(KEY), RequestBody.fromString(CSV_CONTENTS));
+
+        s3Async = S3IntegrationTestBase.s3AsyncClientBuilder().build();
+    }
+
+    @AfterAll
+    public static void teardown() {
+        try {
+            deleteBucketAndAllContents(BUCKET_NAME);
+        } finally {
+            s3Async.close();
+        }
+    }
+
+    @Test
+    public void selectObjectContent_onResponseInvokedWithResponse() {
+        TestHandler handler = new TestHandler();
+        runSimpleQuery(handler).join();
+
+        assertThat(handler.response).isNotNull();
+    }
+
+    @Test
+    public void selectObjectContent_recordsEventUnmarshalledCorrectly() {
+        TestHandler handler = new TestHandler();
+        runSimpleQuery(handler).join();
+
+        RecordsEvent recordsEvent = (RecordsEvent) handler.receivedEvents.stream()
+                                                                         .filter(e -> e.sdkEventType() == EventType.RECORDS)
+                                                                         .findFirst()
+                                                                         .orElse(null);
+
+        assertThat(recordsEvent.payload().asUtf8String()).contains("A\nC");
+    }
+
+    private static CompletableFuture<Void> runSimpleQuery(SelectObjectContentResponseHandler handler) {
```

Comment:
Can we rename this to describe what the query is doing/fetching? This will help give more context to the test assertions.